### PR TITLE
All challenges

### LINKED ISSUE #60: UI Design for All Challenges page
**Here is the final design. Please refer to the screenshots below.**

(There are two screenshots here, of the same page : 
-The first one shows how the page should look if the filter tabs Ongoing and Upcoming challenges are selected, and the Side upper tag showing which one is what(either upcoming or ongoing.
-The second screenshot shows a scenario if all the filter tabs are selected then the side upper tag in each box changes, you can notice the one in the last box changes to the brown color)

evalai_redesign_all_challenges_final_1
evalai_redesign_all_challenges_final_2

**You can refer to the following image for further details on creating the page : **

all-challenge_final_documentation

You can get all the related assets for creating the page from the link below : 
https://drive.google.com/drive/folders/1fhWkj6Ir7HLZzwBZoPCKyyoS-ulFdmg_

The screenshot shows the dummy text, it's just for illustration purpose. You can refer to the current page of the website to find the exact data: https://evalai.cloudcv.org/web/challenges/list

Please comment below if you want to work on this or in case of any question/suggestion.

Mentors for EvalAI-ngx Project (@aka-jain @spyshiv @shivaniprakash95 @RishabhJain2018 @deshraj) will try to respond to this thread ASAP so that you are not blocked.

### PR DESCRIPTION
**Pull request by live-wire**
Supposed to close #60 
This is the complete PR for All Challenges.

As of now, in `My Challenges` displays only the challenges that are `hosted` by a team which the current user is a part of. Should it also display the challenges that the current user is `participating` in ? cc: @deshraj 

Please have a look @RishabhJain2018 @aka-jain @spyshiv 🍕

### CODE REVIEW

**Comment by RishabhJain2018** [comment]
@live-wire Please check why the travis build is failing. 
Moreover,I suggest you to now work in parallel 💻 in order to increase the coverage as current coverage is 59% 😞. We need to increase it to 90% at least 😃 . I am asking this as we haven't merged a lot of features for now so it will be easy to increase the coverage. Once the codebase increases then we won't be able to achieve it and also things might break without tests.
cc: @deshraj @aka-jain @spyshiv

**Comment by live-wire** [comment]
Sure @RishabhJain2018 I'll create an issue to track this and start improving the coverage soon.

**Comment by deshraj** [comment]
Strongly agree with what @RishabhJain2018 said. @live-wire please add tests along with each feature otherwise there is no point in building the features.

**Comment by live-wire** [comment]
@deshraj I agree. I'll add the tests soon. I am building everything as modular/functional as possible. I'll make sure I write good tests too 😄

**Comment by RishabhJain2018** [comment]
@live-wire Ohhh..I missed your first question. Regarding the `My Challenges` part we should have a section for both hosted and participated challenges by a user.
cc: @deshraj

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,2 +1,3 @@
 <!-- The <router-outlet> tells the router where to display routed views. -->
 <router-outlet></router-outlet>
+<app-toast></app-toast>
```
Comment (RishabhJain2018) [inline src/app/app.component.html:3]: Please add a new line here!

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]

**Comment by RishabhJain2018** [comment]
@live-wire please take of this:
screen shot 2018-07-06 at 5 08 29 pm
And challenge details should be publicly accessible to everyone
screen shot 2018-07-06 at 4 59 03 pm

**Comment by live-wire** [comment]
@RishabhJain2018  That Authentication issue is because of fetching the stars which is supposed to change in the API :)

**Comment by RishabhJain2018** [comment]
Closing this since it is already merged.

**Pull request closed without merging**